Source/DafnyLanguageServer/Workspace/DocumentDatabase.cs

```diff
@@ -142,7 +142,11 @@ ISymbolTableRelocator symbolTableRelocator
       if (!VerifyOnSave) {
         return databaseEntry.Document;
       }
-      return VerifyDocumentIfRequiredAsync(databaseEntry);
+      try {
+        return VerifyDocumentIfRequiredAsync(databaseEntry);
+      } catch (OperationCanceledException) {
```

Comment:
This catch-clause won't have any effect since the `OperationCanceledException` is encapsulated in the `Task` returned by `VerifyDocumentIfRequiredAsync`.
You'll need to refactor the method to the async/await pattern of C# and `await` the invocation of `VerifyDocumentIfRequiredAsync`.

Reply:
Good catch. I can confirm that, after removing this change, the the other change is sufficient to fix the issue in my testing.

However, I also agree that the refactoring you describe is cleaner, so I'm going to attempt to get that fully in place today, too. It'll make the `onchange` and `onsave` behavior more symmetrical.